staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -136,10 +138,25 @@ func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
 
 	envVarFiles := os.Getenv(RecommendedConfigPathEnvVar)
 	if len(envVarFiles) != 0 {
-		chain = append(chain, filepath.SplitList(envVarFiles)...)
-
+		files := filepath.SplitList(envVarFiles)
+		for ix := range files {
+			moreFiles, err := loadFileOrDir(files[ix])
+			if err != nil {
+				glog.V(3).Infof("Failed to load config %s (%v)", files[ix], err)
```

Comment:
This splits error handling between two spots and will prevent someone from later adding a "point to directory" option to replicate this functionality.  This logic should be moved to the `Load` function where error handling currently exists and where it would allow for pointing `--kubeconfig` to a directory.